src/io/flutter/view/FlutterViewLocalToggleableAction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.view;
+
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.Presentation;
+import com.intellij.openapi.actionSystem.Toggleable;
+import com.intellij.openapi.application.ApplicationManager;
+import io.flutter.run.daemon.FlutterApp;
+import io.flutter.utils.EventStream;
+import io.flutter.utils.StreamSubscription;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+
+class FlutterViewLocalToggleableAction extends FlutterViewAction implements Toggleable, Disposable {
```

Comment:
Handy class giving an EventStream of toggle button states for buttons that have purely local impacts (e.g. don't need to talk to the Daemon, they just indicate how IntelliJ should render the UI).